src/editor/VideoViewer.js

```diff
@@ -0,0 +1,320 @@
+        this.updateVideoTagMarkup();
+        _viewers[file.fullPath] = this;
+    }
+
+    // Updates the markup used in the preview & the sample markup below
+    VideoView.prototype.updateVideoTagMarkup = function(reload){
+
+        var videoTagAttributesString = "";
+
+        for(var k in this.videoTagSettings) {
+            if(this.videoTagSettings[k]) {
+                videoTagAttributesString = videoTagAttributesString + " " + k;
+                this.$videoEl.attr(k,"");
+            } else {
+                this.$videoEl.removeAttr(k);
+            }
+        }
+
+        var videoTagMarkup =
+            '<video'+ videoTagAttributesString +'>\n' +
+            '  <source src="'+ _getLocalVideoUrl(this.file) + '" type="' + this.videoType + '">\n'+
+            '</video>';
+
+        this.$videoMarkupEl.text(videoTagMarkup);
+
+        // Reloads the video when one of the attributes is changed
+        // so that the preview reflects the changes
+        if(reload) {
+            this.$videoMarkupEl.one("animationend",function(){
+                $(this).removeClass("pop");
+            });
+
+            this.$videoMarkupEl.addClass("pop");
+
+            var videoWrapperHeight = this.$videoWrapperEl.height();
+
+            this.$videoWrapperEl.css("min-height", videoWrapperHeight);
+            this.$videoEl.one("canplay", _.bind(this._onVideoReloaded, this));
+            this.$videoEl[0].load();
+        }
+    };
+
+
+    /**
+     * DocumentManger.fileNameChange handler - when an video is renamed, we must
+     * update the view
+     *
+     * @param {jQuery.Event} e - event
+     * @param {!string} oldPath - the name of the file that's changing changing
+     * @param {!string} newPath - the name of the file that's changing changing
+     * @private
+     */
+    VideoView.prototype._onFilenameChange = function (e, oldPath, newPath) {
+        /*
+         * File objects are already updated when the event is triggered
+         * so we just need to see if the file has the same path as our video
+         */
+        if (this.file.fullPath === newPath) {
+            this.relPath = ProjectManager.makeProjectRelativeIfPossible(newPath);
+        }
+    };
+
+
+    /* Removes min-height property */
+    VideoView.prototype._onVideoReloaded = function (e) {
+        this.$videoWrapperEl.css("min-height", "auto");
+    };
+
+    /**
+     * <video>.on("canplay") handler - updates content of the video view
+     *                            initializes computed values
+     *                            installs event handlers
+     * @param {Event} e - event
+     * @private
+     */
+    VideoView.prototype._onVideoLoaded = function (e) {
+        this._naturalWidth = e.target.videoWidth;
+        this._naturalHeight = e.target.videoHeight;
+
+        var extension = FileUtils.getFileExtension(this.file.fullPath);
+        // TODO: Make a VIDEO_DETAILS string and also add video filesize & length
```

Comment:
Looks like you're doing this below, no?  Also, you can get teh `duration` from the `<video>` element..